### PR TITLE
fix(swap): Use error boundary to handle HW errors

### PR DESCRIPTION
**Pull request by michalszorad**
Resolves [YOMO-873](https://emurgo.atlassian.net/jira/software/c/projects/YOMO/issues/YOMO-873)

[YOMO-873]: https://emurgo.atlassian.net/browse/YOMO-873?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/wallet-mobile/src/features/Swap/common/ModalError/ModalError.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      </View>
+    </>
+  )
+}
+
+const getErrorMessage = (
+  error: Error,
+  strings: Record<'error' | 'rejectedByUser' | 'bluetoothDisabledError', string>,
+): string => {
+  if (error instanceof RejectedByUserError) {
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Swap/common/ModalError/ModalError.tsx:53]: Are there any other errors we should handle?

**Review by jorbuedo — APPROVED** [review]

File: apps/wallet-mobile/src/features/Swap/common/ModalError/ModalError.tsx (hunk 1 above)
Comment (stackchain) [reply apps/wallet-mobile/src/features/Swap/common/ModalError/ModalError.tsx:53]: Please check the MAT funnel.

Comment (michalszorad) [reply apps/wallet-mobile/src/features/Swap/common/ModalError/ModalError.tsx:53]: MAT only shows a general message "Transaction failed". "Your transaction has not been processed properly due to technical issues".

Comment (stackchain) [reply apps/wallet-mobile/src/features/Swap/common/ModalError/ModalError.tsx:53]: https://github.com/Emurgo/yoroi/blob/develop/apps/wallet-mobile/src/yoroi-wallets/hw/hw.ts

https://github.com/Emurgo/yoroi/blob/develop/apps/wallet-mobile/src/yoroi-wallets/cardano/hw/hw.ts

Comment (michalszorad) [reply apps/wallet-mobile/src/features/Swap/common/ModalError/ModalError.tsx:53]: That's now updated

**Pull request merged**